src/alchemlyb/estimators/ti_.py

```diff
@@ -105,3 +105,45 @@ def fit(self, dHdl):
         self.states_ = means.index.values.tolist()
 
         return self
+
+    def separate_dhdl(self):
+        """
+        For transitions with multiple lambda, this function will separate the
+        dhdl with multiple columns into a list of Dataframe with a single column
+        (single lambda).
+        Returns
+        ----------
+        dHdl_list : list
+            A list of Series such that dHdl_list[k][n] is the potential
+            energy gradient with respect to lambda for each configuration n and
+            lambda k.
+        """
+        if len(self.dhdl.index.names) == 1:
+            # If only one column is present convert to series
+            assert len(self.dhdl.columns) == 1
+            name = self.dhdl.columns[0]
+            return [self.dhdl[name], ]
+        else:
```

Comment:
Remove the `else`. The `return` in the previous block terminates control flow. Dedent the following block to make it more readable. It's a bit ugly if the main code in a function is in an `else` block.

Reply:
I read your comment regarding coverage problems. Nevertheless, can you please dedent the whole `else` block and then we see what happens.

Reply:
@xiki-tempula can you please try rewriting without the `else` block?